pkg/platform/kube/functionres/lazy.go

```diff
@@ -581,6 +589,15 @@ func (lc *lazyClient) createOrUpdateDeployment(functionLabels labels.Set,
 			deployment.Spec.Template.Spec.ServiceAccountName = function.Spec.ServiceAccount
 		}
 
+		// enrich deployment spec with default fields that were passed inside the platform configuration
+		// performed on update too, in case the platform config has been modified after the creation of this deployment
+		platformConfigDeployment := lc.platformConfigurationProvider.GetPlatformConfiguration().Kubernetes.Deployment
```

Comment:
Why not make `enrichDeploymentSpec` an `lc` method (Suggest to rename `enrichDeploymentSpecFromPlatformConfiguration`) and move line 557 (=594) inside it?